tests/cecil-tests/GetterExceptionTest.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using Mono.Cecil;
+using System.Collections.Generic;
+using NUnit.Framework;
+using System.Linq;
+using Mono.Cecil.Cil;
+using Xamarin.Utils;
+using Xamarin.Tests;
+
+namespace Cecil.Tests {
+	[TestFixture]
+	public class GetterExceptionTest {
+		readonly HashSet<string> exceptionsToSkip = new ()
+		{
+			"AudioQueueException",
+			"ModelNotImplementedException",
+			"You_Should_Not_Call_base_In_This_Method"
+		};
+
+		bool IsMemberObsolete (ICustomAttributeProvider member)
+		{
+			if (member is null || !member.HasCustomAttributes)
+				return false;
+
+			return member.CustomAttributes.Any ((m) =>
+					m.AttributeType.Name == "ObsoleteAttribute" ||
+					m.AttributeType.Name == "AdviceAttribute" ||
+					m.AttributeType.Name == "ObsoletedOSPlatformAttribute");
+		}
+
+		bool VerifyIfGetterThrowsException (MethodDefinition methodDefinition, out string exceptionMessage)
+		{
+			exceptionMessage = string.Empty;
+			if (methodDefinition?.Body is null)
+				return false;
+
+			foreach (Instruction? inst in methodDefinition.Body.Instructions) {
+				if (inst.OpCode == OpCodes.Newobj) {
```

Comment:
```suggestion
				if (inst.OpCode == OpCodes.Newobj) {
```
You are declaring inst as nullable, becase you are not using ? inst.OpCode can throw a NRE.